numpy/ma/core.pyi

```diff
@@ -1169,14 +1170,27 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
 
     def nonzero(self) -> tuple[_Array1D[intp], ...]: ...
     def trace(self, offset=..., axis1=..., axis2=..., dtype=..., out=...): ...
-    def dot(self, b, out=..., strict=...): ...
+
+    # This differs from `ndarray.dot`, in that 1D dot 1D returns a 0D array.
+    @overload
+    def dot(self, b: ArrayLike, out: None = ..., strict: bool = ...) -> _MaskedArray[Any]: ...  # type: ignore[misc]
```

Comment:
what's the `# type: ignore[misc]` for?

Reply:
nothing thanks, I was hoping some `unused-ignore` error would show up, but running mypy on numpy probably still requires quite some work?